### PR TITLE
Add DF-DCFT warning messages.

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
Adding some warnings regarding #1579.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/dcft/dcft_compute.cc [hunk 1]
```diff
@@ -56,6 +56,9 @@ double DCFTSolver::compute_energy() {
     }
 
     if (options_.get_str("DCFT_TYPE") == "DF") {
+        outfile->Printf(
+            "\n\n\t**** Warning: The density-fitted DCFT cumulant is only approximately variational.\n"
+            "\t     The stationarity conditions, and thus DF-DCFT energy, are subject to change. ****\n");
```
Comment (loriab) [inline psi4/src/psi4/dcft/dcft_compute.cc]: stationary, stationality, stationarity?

Comment (JonathonMisiewicz) [reply psi4/src/psi4/dcft/dcft_compute.cc]: I still prefer "stationarity", but it's not something I'll argue about it if anybody has strong opinions to the contrary.

Comment (loriab) [reply psi4/src/psi4/dcft/dcft_compute.cc]: I don't have strong feelings -- just didn't recognize the word. Ok by me so long as intended, not typo.

Comment (susilehtola) [reply psi4/src/psi4/dcft/dcft_compute.cc]: It should IMHO indeed be "stationarity condition", i.e. "condition for a stationary wave function"

**Pull request merged**